packages/core/src/components/forms/numericInput.tsx

```diff
@@ -353,17 +363,51 @@ export class NumericInput extends AbstractPureComponent<HTMLInputProps & INumeri
     // ===================
 
     private handleDecrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
-        const delta = this.getIncrementDelta(IncrementDirection.DOWN, e.shiftKey, e.altKey);
-        const nextValue = this.incrementValue(delta);
-        this.invokeValueCallback(nextValue, this.props.onButtonClick);
+        this.handleButtonClick(e, IncrementDirection.DOWN);
+    };
+
+    private handleDecrementButtonMouseDown = (e: React.MouseEvent<HTMLInputElement>) => {
+        this.handleButtonClick(e, IncrementDirection.DOWN);
+        this.startContinuousChange();
+    };
+
+    private handleDecrementButtonKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        this.updateDelta(IncrementDirection.DOWN, e);
+    };
+
+    private handleDecrementButtonKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        this.handleButtonKeyUp(e, IncrementDirection.DOWN, this.handleDecrementButtonClick);
     };
 
     private handleIncrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
-        const delta = this.getIncrementDelta(IncrementDirection.UP, e.shiftKey, e.altKey);
+        this.handleButtonClick(e, IncrementDirection.UP);
+    };
+
+    private handleIncrementButtonMouseDown = (e: React.MouseEvent<HTMLInputElement>) => {
+        this.handleButtonClick(e, IncrementDirection.UP);
+        this.startContinuousChange();
+    };
+
+    private handleIncrementButtonKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        this.updateDelta(IncrementDirection.UP, e);
+    };
+
+    private handleIncrementButtonKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        this.handleButtonKeyUp(e, IncrementDirection.UP, this.handleIncrementButtonClick);
+    };
+
+    private handleButtonClick = (e: React.MouseEvent<HTMLInputElement>, direction: IncrementDirection) => {
+        const delta = this.updateDelta(direction, e);
         const nextValue = this.incrementValue(delta);
         this.invokeValueCallback(nextValue, this.props.onButtonClick);
     };
 
+    private handleMouseUp = () => {
+        this.delta = 0;
```

Comment:
this line would make more sense inside `this.stopContinuousChange()`, for completeness